pkg/controllers/routesync/namegeneration.go

```diff
@@ -0,0 +1,72 @@
+// Copied from github.com/openshift/hive/pkg/apis/helpers/namer.go
```

Comment:
This might be useful for other things, so it would be good to move it elsewhere. But we can do that when we actually need it. It's fine here for now.